### PR TITLE
Extending BasiGX.util.Animate

### PR DESCRIPTION
**Pull request by weskamm**
This PR extends the `BasiGX.util.Animation` with a `materialFill` function to animate features.
I have added an example to see the different animations in effect.

### CODE REVIEW

**Comment by weskamm** [comment]
Its ok to merge https://github.com/terrestris/BasiGX/pull/188 before. If this PR breaks after that, i willl fix it

File: examples/animation/animate.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        var interactions = map.getInteractions().getArray();
+        interactions.forEach(function(i) {
+            if (i.getFeatures) {
+                i.getFeatures().clear();
+            }
+        });
+    });
+    hoverSelect.on('select', function(e) {
+        animateFeature(e);
+    });
```
Comment (KaiVolland) [inline examples/animation/animate.js]: Should work without the wrapping function:

`hoverSelect.on('select, animateFeature);`

**Comment by KaiVolland** [comment]
#188 should not be touched by this as the introduced function `materialFill` doesn't use `ol.animation`

**Comment by KaiVolland** [comment]
Feel free to merge.

**Comment by weskamm** [comment]
thanks, merging now

**Pull request merged**